docs/sdks/server-sdks/java.md

```diff
@@ -12,270 +12,374 @@ Eppo's open source Java SDK can be used for both feature flagging and experiment
+Note that `getJSONAssignment()` returns a `JsonNode` from `com.fasterxml.jackson.databind`. If you prefer to use a
+different JSON library, you can use `getJSONStringAssignment()` to get the unparsed JSON string.
 
-If you are using the SDK for experiment assignments, make sure to pass in an assignment logging callback (see the Assignment Logger [section](#define-an-assignment-logger)).
+If you request a type that differs from the flag's variations (for example, you called `getIntegerAssignment()` for a
+flag with string-valued variations), the default value will be returned.
 
+## Define an assignment logger
 
-### Assign variations
+If you are using the Eppo SDK for **experiment** assignments (i.e., randomization), Eppo will need to know which subject,
+(e.g., which user), passed through an entry point and were exposed to the experiment. You will need to log that 
+information to your data warehouse for analysis.
 
-Assig users to flags or experiments using `get<Type>Assignment`, depending on the type of the flag. For example, for a string-valued flag, use `getStringAssignment`:
+When initializing the SDK, define an assignment logger which can handle the `logAssignment()` callback:
 
 ```java
-Optional<String> assignedVariation = eppoClient.getStringAssignment("<SUBJECT-KEY>", "<FLAG-KEY>", {
-  // Optional map of subject metadata for targeting.
-});
+EppoClient.Builder()
+  .apiKey(apiKey)
+  .assignmentLogger(assignmentLogData -> {
+    System.out.println("TODO: send assignment event data to data warehouse: " + assignmentLogData);
+  })
+  .buildAndInit();
 ```
 
-The `getStringAssignment` function takes two required and one optional input to assign a variation:
+The properties of the event object passed to the assignment logger, accessible via getters, are as follows:
 
-- `subjectKey` - The entity ID that is being experimented on, typically represented by a uuid.
-- `flagOrExperimentKey` - This key is available on the detail page for both flags and experiments.
-- `targetingAttributes` - An optional map of metadata about the subject used for targeting. If you create rules based on attributes on a flag/experiment, those attributes should be passed in on every assignment call.
+| Field                                | Description                                                                                                                | Example                                |
+|--------------------------------------|----------------------------------------------------------------------------------------------------------------------------|----------------------------------------|
+| `timestamp` (Date)                   | The time when the subject was assigned to the variation                                                                    | Mon Aug 19 21:46:02 UTC 2024           |
```

Comment:
I imagine the timestamp is not actually `Mon Aug 19 21:46:02 UTC 2024` 😅  -- should we put the actual string value?

Reply:
The JAVA SDK passes `Date` objects around (which hold date and time), so for the example value, I put the `toString()` value of a Java date object.